File: ProjetAttractionSpring/ProjetAttractionBack/src/main/java/attraction/model/Employe.java
package attraction.model;

import javax.persistence.Entity;


@Entity
public class Employe extends Element{

	private String metier;
	private double salaire;
	
	
	public Employe(Integer id, String metier, double salaire) {
		super(id);
		this.metier = metier;
		this.salaire = salaire;
	}


	public Employe(String metier, double salaire) {
		super();
		this.metier = metier;
		this.salaire = salaire;
	}

	
	public Employe() {}
	
	
	public Integer getId() {
		return id;
	}


	public void setId(Integer id) {
		this.id = id;
	}


	public String getMetier() {
		return metier;
	}


	public void setMetier(String metier) {
		this.metier = metier;
	}


	public double getSalaire() {
		return salaire;
	}


	public void setSalaire(double salaire) {
		this.salaire = salaire;
	}


	
	@Override
	public String toString() {
		return "L'employé " + metier + " (id--> " + id+")\n"+"\t|Salaire journalier: " + salaire+ "€\n";
		
	}	
}

File: ProjetAttractionAngular/src/app/model/employe.ts
import { Element } from './element';

export class Employe extends Element{

  constructor(
    private _metier: string,
  	private _salaire: number,
    id?: number
  ){super(id);}


	public get metier(): string {
		return this._metier;
	}

	public set metier(metier: string) {
		this._metier = metier;
	}

	public get salaire(): number {
		return this._salaire;
	}

	public set salaire(salaire: number) {
		this._salaire = salaire;
	}
}
